metagpt/roles/sales.py

```diff
@@ -27,7 +26,7 @@ class Sales(Role):
         "delivered with the professionalism and courtesy expected of a seasoned sales guide."
     )
 
-    store: Optional[BaseStore] = Field(default=None, exclude=True)
+    store: Optional[object] = Field(default=None, exclude=True)  # must inplement tools.SearchInterface
 
     @model_validator(mode="after")
     def validate_stroe(self):
```

Comment:
should SearchEngine be replaced by the Engine in RAG module? it seems we don't need to keep both?

Reply:
This SeachEngine places emphasis on Web Search ability. Replace it with rag or not is the main point of this PR.